### PR TITLE
[DeadCode] Skip RemoveUnusedPrivateMethodRector when class has magic __call

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/skip_magic_call.php.inc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\ClassMethod\RemoveUnusedPrivateMethodRector\Fixture;
+
+/**
+ * @see https://3v4l.org/Lh5O5
+ */
+class SkipMagicCall
+{
+    private function test($params)
```
Comment (staabm) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/skip_magic_call.php.inc:10]: Why should „test“ not be removed.. its still unused?

Comment (samsonasik) [reply rules-tests/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector/Fixture/skip_magic_call.php.inc:10]: I've updated fixture test case to proof it https://github.com/rectorphp/rector/pull/5932/commits/bafe6d097c890ffa2c7f92444b9f8f709936de91 . It called with `$this->__call('test', []);`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php [hunk 2]
```diff
@@ -108,6 +108,10 @@ private function shouldSkip(ClassMethod $classMethod): bool
         }
 
         // skip magic methods - @see https://www.php.net/manual/en/language.oop5.magic.php
-        return $classMethod->isMagic();
+        if ($classMethod->isMagic()) {
+            return true;
+        }
+
+        return $classReflection->hasMethod('__call');
```
Comment (staabm) [inline rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php:115]: Should this checks for invocations of __call() then?

Comment (samsonasik) [reply rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php:115]: `__call` is common used, and running the check can be on specific directory, not the caller, I think currently, this is the safest way.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**